### PR TITLE
MKV/FFV1 tests coherency and automation

### PR DESCRIPTION
**Pull request by JeromeMartinez**
Fixes after automation implementation (checks are unit tested) + few other tests

### CODE REVIEW

**Comment by JeromeMartinez** [comment]
Unit tests:
https://travis-ci.org/MediaArea/MediaConch_SourceCode/jobs/246119798#L2153-L2200

```
OK: Correct_MS_v0.mkv
OK: Correct_MS_v3.mkv
OK: EBML-VER-COH.mkv
OK: EBML-DOCVER-COH.mkv
OK: MKV-ELEMENT-VALID-PARENT.mkv
OK: EBML-ELEMENT-NONMULTIPLES.mkv
OK: EBML-ELEMENT-CONTAINS-MANDATES.mkv
OK: EBML-VALID-MAXID.mkv
OK: EBML-VALID-MAXSIZE.mkv
OK: EBML-ELEMENTS-WITHIN-MAXSIZELENGTH.mkv
OK: EBML-ELEM-UNKNOWN-SIZE.mkv
OK: MKV-SEEK-RESOLVE.mkv
OK: EBML-CRC-VALID.mkv
OK: EBML-ELEMENT-IN-SIZE-RANGE.mkv
OK: EBML-ELEMENT-VALID-RANGE.mkv
OK: MKV-VALID-TRACKTYPE-VALUE.mkv
OK: MKV-NUMERICAL-TAG.mkv
OK: FFV1-HEADER-version-OUTOFBAND_MS_v0.mkv
OK: FFV1-HEADER-version-OUTOFBAND_MS_v3.mkv
OK: FFV1-HEADER-version-LATERVERSION.mkv
OK: FFV1-HEADER-version-EXPERIMENTAL.mkv
OK: FFV1-HEADER-coder_type_MS_v0.mkv
OK: FFV1-HEADER-coder_type_MS_v3.mkv
OK: FFV1-HEADER-state_transition_delta_MS_v0.mkv
OK: FFV1-HEADER-state_transition_delta_MS_v3.mkv
OK: FFV1-HEADER-colorspace_type_MS_v0.mkv
OK: FFV1-HEADER-colorspace_type_MS_v3.mkv
OK: FFV1-HEADER-bits_per_raw_sample_MS_v1.mkv
OK: FFV1-HEADER-bits_per_raw_sample_MS_v3.mkv
OK: FFV1-HEADER-num_h_slices_MS_v3.mkv
OK: FFV1-HEADER-num_v_slices_MS_v3.mkv
OK: FFV1-HEADER-QuantizationTable-len_MS_v3.mkv
OK: FFV1-HEADER-QuantizationTable-scale_MS_v3.mkv
OK: FFV1-HEADER-ec_MS_v3.mkv
OK: FFV1-HEADER-intra_MS_v3.mkv
OK: FFV1-HEADER-configuration_record_crc_parity_v3.mkv
OK: FFV1-SLICE-slice_xywh_h_v3.mkv
OK: FFV1-SLICE-slice_xywh_w_v3.mkv
OK: FFV1-SLICE-slice_xywh_x_v3.mkv
OK: FFV1-SLICE-slice_xywh_y_v3.mkv
OK: FFV1-SLICE-quant_table_index_v3.mkv
OK: FFV1-SLICE-picture_structure_v3.mkv
OK: FFV1-SLICE-slice_crc_parity_v3.mkv
OK: FFV1-SLICE-SliceContent_v3.mkv
OK: FFV1-SLICE-error_status_v3.mkv
OK: FFV1-FRAME-key_frame-ISNOTINTRA_MS_v3.mkv
PASS: test/test_mk.sh
PASS: test/test_ffv1.sh
```

File: matroska/19.md [hunk 1]
```diff
@@ -6,8 +6,15 @@ ASCII-only data in string.
...
 ## Definition
 
-The string element is limited to certain byte ranges of ASCII plus a trailing optional null byte.
+The string element is limited to certain byte ranges of ASCII plus trailing optional null bytes.
 
-## Note
+## Reproducibility
+
+```sh
+./ffmpeg -f lavfi -i "color=c=black:size=2x2" -vframes 1 -pix_fmt gray8 -write_crc32 0 -c:v ffv1 -level 3 -slices 1 /tmp/EBML-ASCII-ONLY-IN-STRING-tmp.mkv
```
Comment (dericed) [inline matroska/19.md]: These mkv outputs are directed to /tmp but other outputs are sent to ./
Also here uses ./ffmpeg and others use ffmpeg. Should be consistent.

File: matroska/19.md [hunk 2]
```diff
@@ -6,8 +6,15 @@ ASCII-only data in string.
...
+xxd -p -c 1000000 /tmp/EBML-ASCII-ONLY-IN-STRING-tmp.mkv > /tmp/EBML-ASCII-ONLY-IN-STRING-tmp.dump
+sed -e 's/9C83756E64/9C83E282AC/g' /tmp/EBML-ASCII-ONLY-IN-STRING-tmp.dump > /tmp/EBML-ASCII-ONLY-IN-STRING-tmp2.dump
+xxd -p -c 1000000 -r /tmp/EBML-ASCII-ONLY-IN-STRING-tmp2.dump > /tmp/EBML-ASCII-ONLY-IN-STRING.mkv
```
Comment (dericed) [inline matroska/19.md]: We use sfk for binary sed in other places. I suggest to replace these four lines of code with two.

```sh
ffmpeg -f lavfi -i "color=c=black:size=2x2" -vframes 1 -pix_fmt gray8 -write_crc32 0 -c:v ffv1 -level 3 -slices 1 EBML-ASCII-ONLY-IN-STRING.mkv
sfk replace EBML-ASCII-ONLY-IN-STRING.mkv -bin '/9C83756E64/9C83E282AC/' -yes
```

File: matroska/19.md (hunk 1 above)
Comment (JeromeMartinez) [reply matroska/19.md]: your modifications are fine (./ffmpeg used only when source code is modified).

**Comment by dericed** [comment]
thx @beyo-ra

**Pull request merged**